cmd/ebcli/main.go

```diff
@@ -0,0 +1,165 @@
+	auth "github.com/cosmos/cosmos-sdk/x/auth/client/rest"
+	bankcmd "github.com/cosmos/cosmos-sdk/x/bank/client/cli"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/client/rest"
+	stakingModule "github.com/cosmos/cosmos-sdk/x/staking"
+
+	stakingclient "github.com/cosmos/cosmos-sdk/x/staking/client"
+	stakingrest "github.com/cosmos/cosmos-sdk/x/staking/client/rest"
+
+	app "github.com/swishlabsco/cosmos-ethereum-bridge"
+	ethbridgeclient "github.com/swishlabsco/cosmos-ethereum-bridge/x/ethbridge/client"
+	ethbridgerest "github.com/swishlabsco/cosmos-ethereum-bridge/x/ethbridge/client/rest"
+)
+
+const (
+	storeAcc       = "acc"
+	routeEthbridge = "ethbridge"
+)
+
+var defaultCLIHome = os.ExpandEnv("$HOME/.ebcli")
+
+func main() {
+	cobra.EnableCommandSorting = false
+
+	cdc := app.MakeCodec()
+
+	// Read in the configuration file for the sdk
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount(sdk.Bech32PrefixAccAddr, sdk.Bech32PrefixAccPub)
+	config.SetBech32PrefixForValidator(sdk.Bech32PrefixValAddr, sdk.Bech32PrefixValPub)
+	config.SetBech32PrefixForConsensusNode(sdk.Bech32PrefixConsAddr, sdk.Bech32PrefixConsPub)
+	config.Seal()
+
+	mc := []sdk.ModuleClients{
+		ethbridgeclient.NewModuleClient(routeEthbridge, cdc),
+		stakingclient.NewModuleClient(stakingModule.StoreKey, cdc),
+	}
+
+	rootCmd := &cobra.Command{
+		Use:   "ebcli",
+		Short: "ethereum bridge client",
+	}
+
+	// Add --chain-id to persistent flags and mark it required
+	rootCmd.PersistentFlags().String(client.FlagChainID, "", "Chain ID of tendermint node")
+	rootCmd.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
+		return initConfig(rootCmd)
+	}
+
+	// Construct Root Command
+	rootCmd.AddCommand(
+		rpc.StatusCommand(),
+		client.ConfigCmd(defaultCLIHome),
+		queryCmd(cdc, mc),
+		txCmd(cdc, mc),
+		client.LineBreak,
+		lcd.ServeCommand(cdc, registerRoutes),
+		client.LineBreak,
+		keys.Commands(),
+		client.LineBreak,
+	)
+
+	executor := cli.PrepareMainCmd(rootCmd, "EB", defaultCLIHome)
+	err := executor.Execute()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func registerRoutes(rs *lcd.RestServer) {
+	rs.CliCtx = rs.CliCtx.WithAccountDecoder(rs.Cdc)
+	rpc.RegisterRoutes(rs.CliCtx, rs.Mux)
+	tx.RegisterRoutes(rs.CliCtx, rs.Mux, rs.Cdc)
+	auth.RegisterRoutes(rs.CliCtx, rs.Mux, rs.Cdc, storeAcc)
+	bank.RegisterRoutes(rs.CliCtx, rs.Mux, rs.Cdc, rs.KeyBase)
+	stakingrest.RegisterRoutes(rs.CliCtx, rs.Mux, rs.Cdc, rs.KeyBase)
+	ethbridgerest.RegisterRoutes(rs.CliCtx, rs.Mux, rs.Cdc, routeEthbridge)
+}
+
+func queryCmd(cdc *amino.Codec, mc []sdk.ModuleClients) *cobra.Command {
+	queryCmd := &cobra.Command{
+		Use:     "query",
+		Aliases: []string{"q"},
+		Short:   "Querying subcommands",
+	}
+
+	queryCmd.AddCommand(
+		rpc.ValidatorCommand(cdc),
+		rpc.BlockCommand(),
+		tx.SearchTxCmd(cdc),
+		tx.QueryTxCmd(cdc),
+		client.LineBreak,
+		authcmd.GetAccountCmd(storeAcc, cdc),
+	)
+
+	for _, m := range mc {
+		mQueryCmd := m.GetQueryCmd()
```

Comment:
this should never return nil right? I'd update to same format as the `txCmd` below

Reply:
fixed in next push